docs/UserGuide.md

```diff
@@ -11,6 +11,50 @@ support engineers, and others can validate a cluster’s configuration against a
 
 # Concepts
 
+## Reference Configuration
+
+The target of `kubectl cluster compare` tool is to detect deviations from a live or local set of CRs with respect to a Reference Configuration specified by a set of Custom Resources (CRs) recommended by Red Hat.
+
+But first, what do we understand by Reference Configuration? The Reference Configuration specifies and filters what your set of CRs looks like. The key to setup the set CRs to specify your Reference Configuration is a YAML file called  `metadata`.
+
+- A Reference Configuration mapped to an OpenShift cluster version. Note that Reference Configurations may change (or not) from one Openshift cluster version to another version.
+- The entity metadata descriptor is a YAML file (`metadata.yaml`) defining an Reference Configuration.
+- A published `metadata.yaml` identifies the set of compliant Reference Configuration CR objects. Any new deviation found after running `kubectl cluster compare` tool shall be identified as a deviation from the expected Reference Configuration.  
+
+- A Reference Configuration mapped to an OpenShift cluster version. Reference Configurations may change (or not) from one Openshift version to another version.
+
+- A Reference Configuration includes an array denoted by `Parts` of one or more objects. Each object includes:
+  - a key "name" with a value of a string typically identifying a workload or a set of workloads
+  - a key "Components" defined as an array of objects
+
+```yaml
+# Every part denotes typically denotes a workload or set of workloads
+ Parts:
+  - name: ExamplePart1 
+    Components:
+      - name: ExampleComponent1
+        ---- here goes ExampleComponent1 configuration ----
+  - name: ExamplePart2
+    ---- here goes Part2 configuration ----
+```
+- "Components" array includes:
+  - a pair at key "name" with a value of a string identifying a component required for the workload:
+  - a pair at key "type" with a value of:
+    - `Optional` for optional components; or
+    - `Required` for mandatory components
+  - the key "requiredTemplates" defined by 
+    - an array of at least one template `YAML` files. Each template `YAML` file has one Reference Configuration CR.
+
+```yaml
+# requiredTemplates contains all needed reference CRs to comply with ExampleComponent1
+Components:
+  - name: ExampleComponent1
+    type: Optional
+    requiredTemplates:
+      - RequiredTemplate1.yaml
+      - RequiredTemplate2.yaml
+```
```

Comment:
I think this level of detail on the structure of the metadata.yaml should go into te reference-config-guide.md which is specifically targeting the people who are creating a reference configuration. There is a section there for Reference CR Groupings which is, I think, the right place for this level of detail.

For a tool user, I do think it is helpful to describe that the reference configuration is broken up into Components which are logical groupings of configuration. Some of these are required and some are optional as defined in the metadata.yaml.

Reply:
ACK. After taking a look at reference-config-guide.md is more suitable than the user guide for this content. Will move that content.